### PR TITLE
LegendasTV: Don't discard provider when BadRarFile or BadZipFile

### PR DESCRIPTION
**Pull request by fernandog**
Show a 'Invalid subtitle' warning instead
Without this, until subtitle is manually downloaded, it will keep discarding provider every run

@ratoaq2

### CODE REVIEW

File: lib/subliminal/providers/legendastv.py [hunk 1]
```diff
@@ -442,4 +442,9 @@ def download_subtitle(self, subtitle):
             self.download_archive(subtitle.archive)
 
         # extract subtitle's content
-        subtitle.content = fix_line_ending(subtitle.archive.content.read(subtitle.name))
+        try:
+            subtitle.content = fix_line_ending(subtitle.archive.content.read(subtitle.name))
+        except (BadRarFile, BadZipfile):
+            # This will make subtitle.is_valid as False and show a 'Invalid Subtitle' warning
+            subtitle.content = None
```
Comment (ratoaq2) [inline lib/subliminal/providers/legendastv.py]: Why do you set content to None?

Comment (fernandog) [inline lib/subliminal/providers/legendastv.py]: Cause none Will make subtitle.is_valid as False. See pytest upstream

Comment (ratoaq2) [inline lib/subliminal/providers/legendastv.py]: But why set None to a field that's already None?

Comment (fernandog) [inline lib/subliminal/providers/legendastv.py]: so what to put in the except ?

Comment (ratoaq2) [inline lib/subliminal/providers/legendastv.py]: A proper log with a proper message so you know what archive is bad

Comment (fernandog) [inline lib/subliminal/providers/legendastv.py]: logger.error('Bad archive for %s', subtitle.name) ?

**Comment by fernandog** [comment]
@ratoaq2 done. Also i was patching the wrong file lol

**Comment by fernandog** [comment]
@ratoaq2 please?

**Pull request merged**